lib/cache/access_request.go

```diff
@@ -0,0 +1,51 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cache
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+func newAccessRequestCollection(upstream services.DynamicAccessCore, w types.WatchKind) (*collection[types.AccessRequest, string], error) {
+	if upstream == nil {
+		return nil, trace.BadParameter("missing parameter DynamicAccess")
+	}
+
+	return &collection[types.AccessRequest, string]{
+		store: newStore(map[string]func(types.AccessRequest) string{
+			"default": func(types.AccessRequest) string { return "default" },
+		}),
+		fetcher: func(ctx context.Context, loadSecrets bool) ([]types.AccessRequest, error) {
+			return nil, nil
+		},
```

Comment:
Sorry, I'm very out of my depth here, but this is somewhat suspicious.

From what I've seen for the store index we usually have something like:

```
		store: newStore(map[proxyServerIndex]func(types.Server) string{
			proxyServerNameIndex: func(u types.Server) string {
				return u.GetName()
			},
		}),
```

Isn't that needed here?

Also why is the fetcher returning nil?

Reply:
Access requests in the cache are weird because the auth doesn't actually cache them but they must be usable in watchers; we should make it more explicit that this is what's happening, or find a better way to deal with resource kinds that are watchable and fanned out but not actually handled by the cache.

Reply:
Some comment could be useful